pkg/config/configmap.go

```diff
@@ -35,13 +37,17 @@ const (
 
 	// clusterCert is the config map key for kourier internal certificates
 	clusterCert = "cluster-cert-secret"
+
+	// timeOut is the amount of time that Kourier waits for incoming requests.
+	IdleTimeoutKey = "stream_idle_timeout"
 )
 
 func DefaultConfig() *Kourier {
 	return &Kourier{
 		EnableServiceAccessLogging: true, // true is the default for backwards-compat
 		EnableProxyProtocol:        false,
 		ClusterCertSecret:          "",
+		IdleTimeout:                0, // infinity
```

Comment:
The current idle timeout is 300, isn't it? If so, we should default to `300` in this PR.

We should carefully change the default to infinity later in another PR as it would affect current users.